src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/DWMColorization.cs

```diff
@@ -19,102 +20,37 @@ internal static Color CurrentApplicationAccentColor
-    private static extern uint GetColorFromPreference(in IMMERSIVE_COLOR_PREFERENCE colorPreference, IMMERSIVE_COLOR_TYPE colorType, bool isHighContrastEnabled, IMMERSIVE_HC_CACHE_MODE cacheMode);
-
-    [StructLayout(LayoutKind.Sequential)]
-    private struct IMMERSIVE_COLOR_PREFERENCE
-    {
-        public uint crStartColor;
-        public uint crAccentColor;
-    }
-
-    private enum IMMERSIVE_COLOR_TYPE
-    {
-        IMCLR_SystemAccentLight1 = 5,
-        IMCLR_SystemAccentLight2 = 6,
-        IMCLR_SystemAccentLight3 = 7,
-        IMCLR_SystemAccentDark1 = 3,
-        IMCLR_SystemAccentDark2 = 2,
-        IMCLR_SystemAccentDark3 = 1,
-        IMCLR_SystemAccent = 4
-    }
-
-    private enum IMMERSIVE_HC_CACHE_MODE
-    {
-        IHCM_USE_CACHED_VALUE = 0,
-        IHCM_REFRESH = 1
-    }
-
     /// <summary>
     /// Gets the system accent color.
     /// </summary>
     /// <returns>Updated <see cref="System.Windows.Media.Color"/> Accent Color.</returns>
     internal static Color GetSystemAccentColor()
     {
-        bool isHighContrastEnabled = SystemParameters.HighContrast;
-      
-        IMMERSIVE_COLOR_PREFERENCE colorPreference = new IMMERSIVE_COLOR_PREFERENCE();
-
-        int err = GetUserColorPreference(in colorPreference, false);
-
-        if (err != 0)
-        {
-            return Color.FromArgb(0xff, 0x00, 0x78, 0xd4);
-        }
-        else
-        {
-            uint AccentColor = GetColorFromPreference(in colorPreference, IMMERSIVE_COLOR_TYPE.IMCLR_SystemAccent, isHighContrastEnabled, IMMERSIVE_HC_CACHE_MODE.IHCM_REFRESH);
-
-            return Color.FromArgb(0xff, (byte)AccentColor, (byte)(AccentColor >> 8), (byte)(AccentColor >> 16));
-        }
+        UISettings _uiSettings3 = new UISettings();
+        return _uiSettings3.GetColorValue(UISettingsRCW.UIColorType.Accent);
     }
 
     /// <summary>
     /// Computes the current Accent Colors and calls for updating of accent color values in resource dictionary
     /// </summary>
     internal static void UpdateAccentColors()
     {
-        Color systemAccent = GetSystemAccentColor();
+        UISettings _uiSettings3 = new UISettings();
+     
+        Color systemAccent = _uiSettings3.GetColorValue(UISettingsRCW.UIColorType.Accent);
+        Color primaryAccent, secondaryAccent, tertiaryAccent;
 
-        Color primaryAccent;
-        Color secondaryAccent;
-        Color tertiaryAccent;
-
-        bool isDarkTheme = ThemeColorization.IsThemeDark();
-        bool isHighContrastEnabled = SystemParameters.HighContrast;
-
-        IMMERSIVE_COLOR_PREFERENCE colorPreference = new IMMERSIVE_COLOR_PREFERENCE();
-
-        int err = GetUserColorPreference(in colorPreference, false);
-
-        if(err != 0) 
+        if(ThemeColorization.IsThemeDark()) 
         {
-            primaryAccent = secondaryAccent = tertiaryAccent = systemAccent;
+            primaryAccent = _uiSettings3.GetColorValue(UISettingsRCW.UIColorType.AccentLight1);
```

Comment:
are you sure you didn't swap light and dark here?

Reply:
if so, it might be worth a comment

Reply:
No we didn't swap it... In dark mode, we use lighter shades of accent color and vice versa.

Yeah, I will add a comment here.

Reply:
Done